src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +82,20 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  public Set<String> getUsers(){
+  Set<String> users = new HashSet<>();
+  Query query = new Query("Message");
+  PreparedQuery results = datastore.prepare(query);
+  for(Entity entity : results.asIterable()) {
+    users.add((String) entity.getProperty("user"));
+  }
+  return users;
+}
+
```

Comment:
nit: Not sure if it is just on my end, but there seems to be a lot of trailing whitespace near the end here.